Sources/NIOHTTP2/HTTP2CommonInboundStreamMultiplexer.swift

```diff
@@ -265,31 +283,97 @@ extension HTTP2CommonInboundStreamMultiplexer {
+    }
+
+    internal func createStreamChannel<Output>(
+        multiplexer: HTTP2StreamChannel.OutboundStreamMultiplexer,
+        promise: EventLoopPromise<Output>?,
+        _ streamStateInitializer: @escaping NIOHTTP2Handler.StreamInitializerWithOutput<Output>
+    ) {
+        if self.channel.eventLoop.inEventLoop {
+            self._createStreamChannel(multiplexer, promise, streamStateInitializer)
+        } else {
+            self.channel.eventLoop.execute {
+                self._createStreamChannel(multiplexer, promise, streamStateInitializer)
+            }
         }
     }
 
-    internal func createStreamChannel(multiplexer: HTTP2StreamChannel.OutboundStreamMultiplexer, _ streamStateInitializer: @escaping (Channel) -> EventLoopFuture<Void>) -> EventLoopFuture<Channel> {
+    internal func createStreamChannel<Output>(
+        multiplexer: HTTP2StreamChannel.OutboundStreamMultiplexer,
+        _ streamStateInitializer: @escaping NIOHTTP2Handler.StreamInitializerWithOutput<Output>
+    ) -> EventLoopFuture<Output> {
+        let promise = self.channel.eventLoop.makePromise(of: Output.self)
+        self.createStreamChannel(multiplexer: multiplexer, promise: promise, streamStateInitializer)
+        return promise.futureResult
+    }
+
+    internal func _createStreamChannel(
+        _ multiplexer: HTTP2StreamChannel.OutboundStreamMultiplexer,
+        _ promise: EventLoopPromise<Channel>?,
+        _ streamStateInitializer: @escaping (Channel) -> EventLoopFuture<Void>
+    ) {
+        let channel = MultiplexerAbstractChannel(
+            allocator: self.channel.allocator,
+            parent: self.channel,
+            multiplexer: multiplexer,
+            streamID: nil,
+            targetWindowSize: Int32(self.targetWindowSize),
+            outboundBytesHighWatermark: self.streamChannelOutboundBytesHighWatermark,
+            outboundBytesLowWatermark: self.streamChannelOutboundBytesLowWatermark,
+            inboundStreamStateInitializer: .excludesStreamID(nil)
+        )
+        self.pendingStreams[channel.channelID] = channel
+        channel.configure(initializer: streamStateInitializer, userPromise: promise)
+    }
+
+    internal func createStreamChannel(
+        multiplexer: HTTP2StreamChannel.OutboundStreamMultiplexer,
+        promise: EventLoopPromise<Channel>?,
+        _ streamStateInitializer: @escaping (Channel) -> EventLoopFuture<Void>
+    ) {
+        if self.channel.eventLoop.inEventLoop {
+            self._createStreamChannel(multiplexer, promise, streamStateInitializer)
+        } else {
+            self.channel.eventLoop.execute {
+                self._createStreamChannel(multiplexer, promise, streamStateInitializer)
+            }
+        }
+    }
+
+    internal func createStreamChannel(
+        multiplexer: HTTP2StreamChannel.OutboundStreamMultiplexer,
+        _ streamStateInitializer: @escaping (Channel) -> EventLoopFuture<Void>) -> EventLoopFuture<Channel> {
```

Comment:
```suggestion
        _ streamStateInitializer: @escaping (Channel) -> EventLoopFuture<Void>
    ) -> EventLoopFuture<Channel> {
```

Reply:
Does this need to be a `NIOHTTP2Handler.StreamInitializer` so we get appropriate `Sendable` annotations?

Reply:
I don’t think that the common one can be `NIOHTTP2Handler.StreamInitializer` because it bubbles up to the legacy `StreamMultiplexer` create stream which is public and doesn’t have `Sendable` annotations.